src/backend/distributed/utils/citus_depended_object.c

```diff
@@ -490,6 +491,21 @@ AnyObjectViolatesOwnership(DropStmt *dropStmt)
 		}
 		PG_CATCH();
 		{
```

Comment:
This thing seems a bit dodgy. The proper way to continue after a PG_TRY that read from the database (catalog) would be to use a subtransaction, such that memory and other resources are properly cleaned and we do not leave behind e.g. a half-initialized cache. However, that'll trigger Citus to emit savepoints to worker nodes.

It would be very much preferable to write a non-throwing variant of check_object_ownership.

Reply:
Subtransaction seems a better option since there is no non-throwing variant for the ownership check. It could be hard to maintain if we make it our codebase.

Fortunately, Citus does not emit savepoints because we will never be in coordinated transaction. (we do ownership checks before running preprocess and postprocess steps to disable them when the checks fail)